packages/block-editor/src/components/block-list/content.scss

```diff
@@ -360,6 +360,19 @@ _::-webkit-full-page-media, _:future, :root .has-multi-selection .block-editor-b
 	z-index: z-index("{core/image aligned left or right} .wp-block");
 }
 
+// This could very well break themes. I wonder if we should just set body
+// element height, and let themes decide if they want to push the footer to the
+// bottom.
+.is-root-container.wp-site-blocks {
+	display: flex;
+	flex-direction: column;
+	height: 100%;
+
+	main {
+		flex: 1;
+	}
+}
```

Comment:
@jasmussen @MaggieCabrera Sorry to ping you, but maybe with your CSS wizardry skills you can think of ways to make this work for all themes. Otherwise I think we'll have to rely on themes to add it. Or maybe this is fine for block-based themes if they don't use a custom stylesheet?

Reply:
Can we make this css conditional to having no patterns inserted instead? We only care about this when the page is empty, otherwise if we insert a pattern that doesn't fill the whole viewport height we are introducing fake space that is not really there in the frontend.

Reply:
Sorry I'm catching up after a bit. What does this property do? 

I'd agree we want to ideally avoid a theme needing to add anything at all in order for the zoom prop to work.

Reply:
How does it break themes? Doesn't this CSS be just in editor? Also what makes it work in zoom out mode only?

Reply:
I think there's some misunderstandings:

* No, this CSS is not required for zoom out mode, it's just nice that the footer is pushed down when the viewport is made higher.
* Since it's not a requirement for zoom out mode, it something that we let theme implement. I think this is a nice thing on the front-end as well (pushing the footer down when there's no or little content).
* I don't see this as fake space at all, I just see this ensuring the footer is at the bottom of the page. Again, if it's something that the theme does it would be on the front-end too.
* I didn't say it breaks themes, I just said it may clash with theme CSS, but actually this may not be the case at all for block based themes 🤔 

Or maybe we can rely on block-based themes never styling `is-root-container`, in which case this CSS may be safe to use as it won't clash with anything. It still relies on `header`, `main` and `footer` being top-level blocks though.

Reply:
> No, this CSS is not required for zoom out mode, it's just nice that the footer is pushed down when the viewport is made higher.

Oh, can you clarify this a bit more? Ideally the shape of the page doesn't change when you zoom out. If the page is short, it should be even shorter when zoomed out.

Reply:
Valid enough, I think there's some nuance to explore there in terms of what that means for the blue pattern insertion plus buttons as well, and where they show up. I.e. to your point, we certainly don't want those to stack right next to each other. But we also want to avoid an accordion behavior where the document is always _at least full-viewport-height_ when empty, and then it collapses as soon as you add a single section pattern. So I'd be tempted to apply a _min-height_ instead of flex magic.

Reply:
@jasmussen I think this does exactly what you say: https://github.com/WordPress/gutenberg/pull/59512#pullrequestreview-1919411000

Reply:
> Ideally the shape of the page doesn't change when you zoom out. If the page is short, it should be even shorter when zoomed out.

The whole point of this PR is to not have that. Maybe @richtabor can explain this better :)

Reply:
Just to clarify, this simply ensures the footer is at the bottom of an empty page, or rather when there is not enough content to push it there—and only when zoom-out mode is engaged. This is not a "fit" view.

It helps align the interface with expectations of where content (i.e. patterns) will be inserted. 

| Before  | After |
| ------------- | ------------- |
|![CleanShot 2024-03-07 at 14 56 27](https://github.com/WordPress/gutenberg/assets/1813435/044edaee-c3b0-427a-856c-ff3c1bc8b04f)|![CleanShot 2024-03-07 at 14 55 51](https://github.com/WordPress/gutenberg/assets/1813435/b7de9c78-2076-468c-acf9-e9328bbdf2be)|